### PR TITLE
feat(sidecar): Sidecar for asciidoc plugins that contains tools like asciidoc

### PR DESCRIPTION
**Pull request by sunix**
Signed-off-by: Sun Seng David TAN <sutan@redhat.com>

### What does this PR do?
Provide a sidecar for the existing asciidoc plugin so task to export to PDF works well.
The sidecar contains asciidoc-topdf

### Screenshot/screencast of this PR
Selection_114

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/16370

### How to test this PR?
- Use this devfile (that uses the sidecar built and pushed to my quay.io repo):
```yaml
apiVersion: 1.0.0
metadata:
  generateName: ascii-doc-
projects:
  - name: asciidoctor-che
    source:
      location: 'https://github.com/sunix/asciidoctor-che'
      type: git
      branch: master
components:
  - id: joaompinto/asciidoctor-vscode/latest
    type: chePlugin
    registryUrl: 'https://sunix-che-plugin-registry-dev-asciidoc.surge.sh/v3'
  - mountSources: true
    args:
      - sleep
      - infinity
    memoryLimit: 768Mi
    type: dockerimage
    image: quay.io/sunix/asciidoctor
    alias: asciidoctor
commands:
  - name: asciidoctor README.adoc
    actions:
      - workdir: /projects/asciidoctor-che
        type: exec
        command: asciidoctor README.adoc
        component: asciidoctor
  - name: asciidoctor-pdf README.adoc
    actions:
      - workdir: /projects/asciidoctor-che
        type: exec
        command: asciidoctor-pdf README.adoc
        component: asciidoctor
  - name: asciidoctor-epub3 README.adoc
    actions:
      - workdir: /projects/asciidoctor-che
        type: exec
        command: asciidoctor-epub3 README.adoc
        component: asciidoctor
```
- open asciidoc-che/README.adoc
- `F1` -> `Asciidoc: Export document to pdf`
- Choose a location
- The PDF file has been generated

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [x] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [x] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [x] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [x] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [x] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [x] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [x] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [x] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

**Comment by vinokurig** [comment]
@sunix Why not to add the sidecar to the plugin by default?

**Comment by sunix** [comment]
> @sunix Why not to add the sidecar to the plugin by default?

It is the second step, the sidecar needs to be in master so the plugin registry generator can pick it.

File: sidecars/asciidoc/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM asciidoctor/docker-asciidoctor
```
Comment (svor) [inline sidecars/asciidoc/Dockerfile:1]: I think it should contain license as well

Comment (sunix) [reply sidecars/asciidoc/Dockerfile:11]: @svor done :+1:

**Review by vinokurig — APPROVED** [review]
Works as described in the **How to test** section.

**Review by svor — APPROVED** [review]
screenshot-che openshift io-2020 12 23-10_52_11

**Pull request merged**